tests/test_create.py

```diff
@@ -2484,24 +2523,33 @@ def side_effect(self, url, **kwargs):
 
                         SubdirData.clear_cached_local_channel_data(exclude_file=False)
 
-                        # This first install passes because flask and its dependencies are in the
-                        # package cache.
                         assert not package_is_installed(prefix, "flask")
-                        run_command(
+                        command = (
                             Commands.INSTALL,
                             prefix,
                             "-c",
                             channel,
                             "flask",
                             "--offline",
                         )
-                        assert package_is_installed(prefix, "flask")
-
-                        # The mock should have been called with our local channel URL though.
-                        assert result_dict.get("local_channel_seen")
+                        if context.solver == "libmamba":
+                            # libmamba solver expects repodata to be loaded into Repo objects
+                            # It doesn't use the info from the tarball cache as conda does
+                            with pytest.raises((RuntimeError, UnsatisfiableError)):
+                                run_command(*command)
+                        else:
+                            # This first install passes because flask and its dependencies are in the
+                            # package cache.
+                            run_command(*command)
+                            assert package_is_installed(prefix, "flask")
+
+                            # The mock should have been called with our local channel URL though.
+                            assert result_dict.get("local_channel_seen")
 
                         # Fails because pytz cannot be found in available channels.
-                        with pytest.raises(PackagesNotFoundError):
+                        # TODO: conda-libmamba-solver <=23.9.1 raises an ugly RuntimeError
+                        # We can remove it when 23.9.2 is out with a fix
```

Comment:
The fix mentioned in the comment is https://github.com/conda/conda-libmamba-solver/pull/323